src/storage-client/src/controller.rs

```diff
@@ -1939,12 +1940,13 @@ where
                         .extend(update.iter().cloned());
                 }
 
-                let (changes, frontier) = storage_net
+                let (changes, frontier, cluster_id) = collections_net
                     .entry(key)
-                    .or_insert_with(|| (ChangeBatch::new(), Antichain::new()));
+                    .or_insert_with(|| (ChangeBatch::new(), Antichain::new(), None));
 
                 changes.extend(update.drain());
                 *frontier = collection.read_capabilities.frontier().to_owned();
+                *cluster_id = collection.cluster_id();
```

Comment:
One more question, why is this not part of the `or_insert_with(..)` call? In which situation the collection doesn't have the correct cluster from the beginning?

Reply:
No good reason, I just modeled it on the other fields. Changing :ok_hand: